From 00b1d82a5ea9760bfc0fc5d280f479d776c0013f Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Sun, 22 Jan 2017 11:23:01 +0200
Subject: [PATCH 0824/1345] fix: dts: a8040: add missing MPP configuration for
 AP SPI

commit  345125d27a38d8f340af208a06e17bc51348d7ea from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add AP SPI pins configuration to armada-8040.dtsi.
Add binding between AP SPI pins to AP SPI node in
armada-8040-db-E.dts.

Change-Id: Iba125dafd1382fc7cd6c4c67a6fb90ec5e1cdfe0
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35865
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37331
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi     |    5 +++++
 2 files changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index 8c0c18c..8d30144 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -70,6 +70,8 @@
 			};
 			spi0: spi@510600 {
 				status = "okay";
+				pinctrl-names = "default";
+				pinctrl-0 = <&ap_spi0_pins>;
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 488c342..04604c6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -124,6 +124,11 @@
 					"mpp8", "mpp9", "mpp10";
 					marvell,function = "sdio";
 				};
+				ap_spi0_pins: spi-pins-0 {
+					marvell,pins = "mpp0", "mpp1", "mpp2",
+					"mpp3";
+					marvell,function = "spi0";
+				};
 			};
 		};
 	};
-- 
1.7.9.5

